src/MCParticleManager.cc

```diff
@@ -262,19 +262,23 @@ std::set<G4PrimaryParticle*> MCParticleManager::createPrimary(G4PrimaryParticle*
 #endif
     }
 
-    for (int i = 0; i < mcp->getDaughters().size(); i++) {
+    if (_recursionBlockSet.count(mcp) == 0) {
```

Comment:
This is where exclusion occurs.  This breaks recursion if the daughters were already scanned.